server/services/admin.ts

```diff
@@ -210,18 +217,97 @@ export = ({ strapi }: StrapiContext): IServiceAdmin => ({
     };
   },
 
+  //Find all reports
+  async findReports(
+    this: IServiceAdmin,
+    query: AdminFindAllProps,
+  ): Promise<ToBeFixed> {
+    const {
+      _q,
+      filters,
+      orderBy,
+      page = 1,
+      pageSize = 10,
+    }: AdminFindAllQueryParamsParsed = parseParams<AdminFindAllQueryParamsParsed>(
+      query,
+    );
+
+    const defaultWhere = {
+      resolved: { $notNull: true },
+    };
+
+    const defaultAuthorUserPopulate = this.getDefaultAuthorPopulate();
+
+    let params: StrapiDBQueryArgs<CommentModelKeys> = {
+      where: !isEmpty(filters)
+        ? {
+            ...defaultWhere,
+            ...filters,
+          }
+        : { ...defaultWhere },
+      offset: (page - 1) * pageSize,
+      limit: pageSize,
+      orderBy: orderBy || [{ resolved: "asc" }, { createdAt: "desc" }],
+    };
+    if (_q) {
+      params = {
+        ...params,
+        where: {
+          ...params.where,
+          content: {
+            $contains: _q,
+          },
+        },
+      };
+    }
+
+    const entities = await strapi.db
+      .query<Comment>(getModelUid("comment-report"))
+      .findMany({
+        ...params,
+        populate: ["related"],
+      });
+
+    const total = await strapi.db
+      .query<Comment>(getModelUid("comment-report"))
+      .count({
+        where: params.where,
+      });
+
+    const result = entities.map((_) =>
+      filterOurResolvedReports(
+        this.getCommonService().sanitizeCommentEntity(
+          _,
+          defaultAuthorUserPopulate?.populate,
+        ),
+      ),
+    );
+
+    const pageCount = Math.floor(total / pageSize);
+
+    return {
+      message: "Hello there!",
+      result,
+      pagination: {
+        page: page,
+        pageSize: pageSize,
+        pageCount: total % pageSize === 0 ? pageCount : pageCount + 1,
+        total,
+      },
+    };
+  },
+
   // Find single comment
   async findOneAndThread(
     this: IServiceAdmin,
     id: Id,
-    { removed, ...query }: AdminFindOneAndThreadProps
+    { removed, ...query }: AdminFindOneAndThreadProps,
   ): Promise<AdminSinglePageResponse> {
-
     const defaultAuthorUserPopulate = this.getDefaultAuthorPopulate();
 
     const defaultWhere = !removed
       ? {
-          $or: [{ removed: false }, { removed: null }],
+          $or: [{ removed: false }, { removed: { $notNull: true } }],
```

Comment:
All new comments have `removed: null`, in this case no data will be fetched without filters set